### ISSUE TITLE
Fix bug (blocking) in new SQLAlchemy migration tests

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#2404]
Currently migration tests risk to block (and not even CTRL+C is enough to stop, one needs to kill the process).

The reason is that sessions are not properly closed in the new test environment, so the lock to ALTER the table in the migration cannot be acquired and Postgres hangs.

**Comment by giovannipizzi** [comment]
I just report here a few useful SQL commands to be executed in psql that I found useful to check what was going on and what was locked, for future reference
```
SELECT pid, wait_event, wait_event_type, datname, query_start, state, LEFT(query, 20) FROM pg_stat_activity;
  pid  | wait_event | wait_event_type |  datname  |          query_start          |        state        |         left         
-------+------------+-----------------+-----------+-------------------------------+---------------------+----------------------
 77988 |            |                 | test_sqla | 2019-01-21 15:21:45.600842+01 | idle in transaction | SELECT db_dbgroup.id
 77998 |            |                 | test_sqla | 2019-01-21 15:21:45.839774+01 | idle                | COMMIT
 78001 |            |                 | test_sqla | 2019-01-21 15:21:45.81475+01  | idle in transaction | SELECT db_dbgroup.id
 78002 | relation   | Lock            | test_sqla | 2019-01-21 15:21:45.861368+01 | active              | ALTER TABLE db_dbgro
 78264 |            |                 | test_sqla | 2019-01-21 15:33:24.799716+01 | active              | SELECT pid, wait_eve
(5 rows)
```

```
SELECT relation::regclass, * FROM pg_locks WHERE NOT GRANTED;
  relation  | locktype | database | relation | page | tuple | virtualxid | transactionid | classid | objid | objsubid | virtualtransaction |  pid  |        mode         | granted | fastpath 
------------+----------+----------+----------+------+-------+------------+---------------+---------+-------+----------+--------------------+-------+---------------------+---------+----------
 db_dbgroup | relation |   116575 |   180736 |      |       |            |               |         |       |          | 5/6                | 78002 | AccessExclusiveLock | f       | f
(1 row)
```

```
SELECT * FROM pg_stat_activity WHERE wait_event != '';
```

```
SELECT * FROM pg_locks WHERE pid = 78002 and NOT granted;
```

**Comment by ConradJohnston** [comment]
Fixed in #2405
